src/firefly/js/util/VOAnalyzer.js

```diff
@@ -0,0 +1,337 @@
+    constructor(tableModel, posCoord='eq') {
+        this.tableModel = tableModel;
+        this.columns = get(tableModel, ['tableData', 'columns'], []);
+        this.obsCoreInfo = {isChecked: false, isObsCoreTable: false};
+        this.posCoord = posCoord;
+        this.centerColumnsInfo = null;
+    }
+
+    isObsCoreTable() {
+        if (this.obsCoreInfo.isChecked) {
+            return this.obsCoreInfo.isObsCoreTable;
+        }
+        const allColNames = this.columns.map((oneCol) => oneCol.name);
+
+        const nonExistCol = ObsTapMandColumns
+                            .map((oneColumn) => (oneColumn[ObsTapColName]))
+                            .some((oneName) => {
+                                return !allColNames.includes(oneName);
+                            });
+
+        this.obsCoreInfo.isChecked = true;
+        this.obsCoreInfo.isObsCoreTable = !nonExistCol;
+
+        return this.obsCoreInfo.isObsCoreTable;
+    }
+
+    getCenterColumnsInfo() {
+        return isEmpty(this.centerColumnsInfo) ? null : this.centerColumnsInfo;
+    }
+
+    setCenterColumnsInfo(lonCol, latCol, lonIdx, latIdx, csys = CoordinateSys.EQ_J2000) {
+        this.centerColumnsInfo = {
+            lonCol,
+            latCol,
+            lonIdx,
+            latIdx,
+            csys
+        };
+    }
+
+    /**
+     * get columns containing the same ucd value
+     * @param ucd
+     * @returns {Array}
+     */
+    getTblColumnsOnUCD(ucd) {
+        return this.columns.reduce((prev, oneCol) => {
+            if (oneCol.UCD && oneCol.UCD.includes(ucd)) {
+                prev.push(oneCol);
+            }
+            return prev;
+        }, []);
+    }
+
+    /**
+     * get center columns pairs by checking ucd values
+     * @param coord
+     * @returns {Array}  [[pair_1_col_ra, pair_1_col_dec], ...., [pair_n_col_ra, pair_n_col_dec]]
+     */
+    getCenterColumnPairsOnUCD(coord = this.posCoord || UCDCoord.eq.key) {
+        const centerColUCDs = has(posCol, coord ) ? posCol[coord].ucd : null;
+        const pairs = [];
+
+        if (!centerColUCDs) {
+            return pairs;
+        }
+
+        // get 'ra' column list and 'dec' column list
+        let posPairs = centerColUCDs.reduce((prev, eqUcd) => {
+            const cols = this.getTblColumnsOnUCD(eqUcd);
+
+            prev.push(cols);
+            return prev;
+        }, []);
+
+        // no pair
+        if (posPairs[0].length === 0 || posPairs[1].length === 0) {
+            return pairs;
+        }
+
+        // only one pair
+        if (posPairs[0].length === 1 && posPairs[1].length === 1) {
+            pairs.push([posPairs[0][0], posPairs[1][0]]);
+        } else {
+            const mainColIdx = [];
+
+            // find the leftmost pair with 'main.meta'
+            for (let n = 0; n < 2; n++) {
+                mainColIdx.push(posPairs[n].findIndex((col) => {
+                    return (col.UCD && col.UCD === `${centerColUCDs[n]};${mainMeta}`);
```

Comment:
_A UCD is a string which contains textual tokens called ‘words’, separated by semicolons(;). A word is composed of ‘atoms’, separated by periods(.)._
We should not rely on the certain order of UCD words when detecting the presence of a particular word in UCD. They can be in any order. We might want to split UCD by semicolon and then check for the presence of a particular word.

Reply:
per UCD document, http://www.ivoa.net/documents/REC/UCD/UCDlist-20070402.pdf, the token's position might be fixed as first word or secondary word. I will exam the code again to make sure the 'if condition' has the right test logic.